### PR TITLE
Add BUIDLHub 'Email' reminder option to Address / Name pages.

### PR DESCRIPTION
**Pull request by mjpowersjr**
Adds a new 'Email' reminder option to the Name & Address pages. Introduces a single additional dependency on a HOC component to manage communication with https://buidlhub.com 

Source available at: https://github.com/BUIDLHub/buidlhub-ens-notifications

## Issue number
N/A

## Description
Bulk of changes have been isolated to new components added to `src/components/ExpiryNotification`. Some minor changes where required to integrate with the add-to-calendar-hoc component. A few additional translation keys have been provided (English only) as well.

## List of features added/changed

- Add dependency on `@buidlhub/buidlhub-ens-notifications`
- Add Email notification modals
- Minor adjustments to impacted components / pages for integration.

## How Has This Been Tested?

Performed static build of the site, performed manual testing. Static build can (temporarily) be found at:

http://ens.demo.buidlhub.com.s3-website-us-east-1.amazonaws.com/

## Screenshots (if appropriate):
N/A

## Checklist:

- [X] My code follows the code style of this project.
- [X] My code implements all the required features.

### CODE REVIEW

File: public/locales/en.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "Address": "Address",
     "Content": "Content",
     "Controller": "Controller",
+    "email": "Email",
```
Comment (makoto) [inline public/locales/en.json:56]: Please duplicate the same in `public/locals/en.json`.

You can just run `yarn i18n` to generate it

File: public/locales/en.json [hunk 2]
```diff
@@ -126,6 +127,9 @@
     },
     "subdomainFavouriteTitle": "Favourite Subdomains"
   },
+  "expiry": {
+    "reminder": "Set Renewal Reminder"
+  },
```
Comment (makoto) [inline public/locales/en.json:132]: Please duplicate the same in `public/locals/en.json`.

You can just run `yarn i18n` to generate it

File: src/components/Address/Address.js [hunk 3]
```diff
@@ -250,14 +252,18 @@ export default function Address({
...
           />
-          <Sorting
-            activeSort={activeSort}
-            setActiveSort={setActiveSort}
-            activeFilter={domainType}
-          />
 
-          {domainType === 'registrant' && (
-            <>
+          <div
```
Comment (makoto) [inline src/components/Address/Address.js:256]: Please refrain from inlining css styling. Please follow [this guide ](https://github.com/ensdomains/ens-app#cssstyling) for more detail.

File: src/components/ExpiryNotification/EmailNotifyLink.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// import styled from '@emotion/styled'
```
Comment (makoto) [inline src/components/ExpiryNotification/EmailNotifyLink.js:1]: Please remove unused comment

File: src/components/ExpiryNotification/ExpiryNotificationModal.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  display: grid;
+  grid-gap: 10px;
+  grid-template-columns: 1fr;
+  ${mq.xLarge`
+    grid-template-columns: 2fr 1fr;
+  `}
+`
+
+const LabelComponent = ({ address, name }) => {
+  // const { t } = useTranslation()
```
Comment (makoto) [inline src/components/ExpiryNotification/ExpiryNotificationModal.js:48]: Please remove unused comment

File: src/components/ExpiryNotification/ExpiryNotifyDropdown.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+// import { motion, AnimatePresence } from 'framer-motion'
```
Comment (makoto) [inline src/components/ExpiryNotification/ExpiryNotifyDropdown.js:3]: Please remove unused comment

File: src/components/ExpiryNotification/TruncatedAddress.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+export default function TruncatedAddress({ address = '' }) {
+  // const [showFullAddress, setShowFullAddress] = useState(false)
```
Comment (makoto) [inline src/components/ExpiryNotification/TruncatedAddress.js:4]: Is this commented out for some sort of debugging purpose?

**Comment by makoto** [comment]
Screenshot 2020-05-26 at 15 20 34

2 styling issues on the modal.

- Please add more padding to the left side of email placeholder message.
- There is a grey bar on the right side

**Comment by makoto** [comment]
On mobile view, the email reminder link should be in separate line so that we can see renewal box.
Screenshot 2020-05-26 at 15 25 00

**Comment by makoto** [comment]
Screenshot 2020-05-26 at 15 26 43

Renew window should expand to the left but it got blocked by the renewal link

**Comment by makoto** [comment]
Screenshot 2020-05-26 at 15 29 56

The confirmation modal seems to have different design

File: public/locales/en.json (hunk 1 above)
Comment (mjpowersjr) [reply public/locales/en.json:56]: Running `yarn i18n` seems to add the corresponding translation to `public/locals/cn.json` - This leaves a English translation in the `cn.json` file, is that ok? 

Results from runniny `yarn i18n`:
`public/locales/cn.json`
```json
...
    "edit": "编辑",
    "email": "Email",
    "expires": "到期",
...
```

I'm certainly not qualified to do an actual CN translation for the site. :-)

File: public/locales/en.json [hunk 8]
```diff
@@ -126,6 +127,9 @@
     },
     "subdomainFavouriteTitle": "Favourite Subdomains"
   },
+  "expiry": {
+    "reminder": "Set Renewal Reminder"
```
Comment (makoto) [inline public/locales/en.json:131]: Please change it to `Remind me`

File: src/components/ExpiryNotification/EmailNotifyLink.js (hunk 4 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/EmailNotifyLink.js]: Removed

File: src/components/ExpiryNotification/ExpiryNotificationModal.js (hunk 5 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js]: Removed

File: src/components/ExpiryNotification/ExpiryNotifyDropdown.js (hunk 6 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotifyDropdown.js]: Removed

File: src/components/ExpiryNotification/TruncatedAddress.js (hunk 7 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/TruncatedAddress.js]: Removed - Originally I had provided the ability to expand a truncated address by clicking on the ellipsis, but this caused layout issues within the modal. I left this in place in-case we wanted to revisit that capability later.

File: public/locales/en.json (hunk 8 above)
Comment (mjpowersjr) [reply public/locales/en.json]: Updated to `Remind Me`

**Comment by mjpowersjr** [comment]
> There is a grey bar on the right side

I'm looking into this, in Chrome I get a read-only scroll bar, while in Firefox I get a dark gray bar, similar to the one in your screenshot (although it does look slightly different).

**Comment by mjpowersjr** [comment]
> > There is a grey bar on the right side
> 
> I'm looking into this, in Chrome I get a read-only scroll bar, while in Firefox I get a dark gray bar, similar to the one in your screenshot (although it does look slightly different).

ok, it looks like the scroll bar / gray on the right hand side of the modal was not introduced as part of this PR. The screenshot below is from the current version of the ENS site. How would you like to precede with regards to the gray bar?

ens-modal-scroll-exampe

**Comment by mjpowersjr** [comment]
> The confirmation modal seems to have different design

* I've adjusted the font-weight to match the design. (not pushed up yet)
* What about the lack of `Close` button in the original design - Do you want the button incorporated or removed?

File: public/locales/en.json (hunk 1 above)
Comment (makoto) [reply public/locales/en.json]: yes, that's expected. It's more for the place holder.

**Comment by makoto** [comment]
> How would you like to precede with regards to the gray bar?

If it's same in other modals, then let's just keep it as is.

**Comment by makoto** [comment]
> What about the lack of Close button in the original design

I think most people intuitively click outside of the modal which closes the popup.

**Comment by makoto** [comment]
One more thing.

In mobile view, I cannot see input field border.

Screenshot 2020-05-26 at 21 34 23

File: src/components/ExpiryNotification/ExpiryNotificationModal.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+  // feature would require additional planning / coordination between ENS <> BUIDLHub.
+
+  // const domainName = currentModal ?
+  //   currentModal.domainName :
+  //   null;
+
+  const translation = {
+    cancel: t(`c.cancel`),
+    submit: t(`c.save`),
+    close: 'Close',
```
Comment (jefflau) [inline src/components/ExpiryNotification/ExpiryNotificationModal.js:112]: Please add translation strings for all of these `en.json`

Comment (jefflau) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js:112]: And after please run `yarn i18n` to duplicate the keys in non English languages

Comment (jefflau) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js:112]: Looks like "Close" would be good in `c.close` and the rest would need a namespace such as `emailReminder.placeholder`

File: src/components/ExpiryNotification/ExpiryNotificationModal.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    placeholder: 'Enter Email address',
+    registerSuccess:
+      'Please check your inbox to verify your email address. You will be redirected to BUIDLHub to manage your email notifications.'
+  }
+
+  return (
+    <Modal name={EXPIRY_NOTIFICATION_MODAL_NAME}>
+      <EmailComponent
+        cancelComponent={CancelComponent}
+        // domainName={domainName}
```
Comment (jefflau) [inline src/components/ExpiryNotification/ExpiryNotificationModal.js:122]: Do we still need this?

File: src/components/Calendar/Calendar.js [hunk 11]
```diff
@@ -48,15 +57,30 @@ function CalendarInvite({ startDatetime, type = '', name, noMargin }) {
...
+    </EmailNotifyLink>
+  )
+
   const AddToCalendar = AddToCalendarHOC(CalendarButton, Dropdown)
   return (
     <AddToCalendar
       event={event}
       className={styles}
-      buttonText="Set renewal reminder"
+      buttonText={t('expiry.reminder')}
```
Comment (jefflau) [inline src/components/Calendar/Calendar.js:75]: Change this to a more specific namespace such as `expiryNotification` and add the other keys there too

File: src/components/Address/Address.js [hunk 12]
```diff
@@ -250,14 +252,18 @@ export default function Address({
...
+            className={css`
```
Comment (jefflau) [inline src/components/Address/Address.js:257]: Let's change this to a styled component. I've used it once or twice in the project before, but generally we use styled-components to reduce the bloat in the render function

File: src/components/ExpiryNotification/ExpiryNotificationModal.js [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+  ${mq.xLarge`
+    grid-template-columns: 2fr 1fr;
+  `}
+`
+
+const LabelComponent = ({ address, name }) => {
+  return (
+    <>
+      <div
+        className={css`
```
Comment (jefflau) [inline src/components/ExpiryNotification/ExpiryNotificationModal.js:51]: Change to a styled-component

File: src/components/ExpiryNotification/ExpiryNotificationModal.js [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+        `}
+      >
+        <a href={BUILDHUB_LINK} target="_blank" rel="noopener noreferrer">
+          BUIDLHub
+        </a>{' '}
+        checks if address <TruncatedAddress address={address} /> has expiring
+        names within 30 days and sends you a reminder email every week.
+        <label
+          htmlFor={name}
+          className={css`
```
Comment (jefflau) [inline src/components/ExpiryNotification/ExpiryNotificationModal.js:62]: Change to a styled component

File: src/components/ExpiryNotification/TruncatedAddress.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+export default function TruncatedAddress({ address = '' }) {
+  const showFullAddress = false
+  const addressTruncated = address.substr(0, 14)
+
+  return (
+    <span title={address} style={{ fontWeight: 500 }}>
```
Comment (jefflau) [inline src/components/ExpiryNotification/TruncatedAddress.js:8]: Definitely don't want any javascript styles if possible, please change to a styled-component

**Comment by mjpowersjr** [comment]
> * Please add more padding to the left side of email placeholder message.

Fixed!

File: src/components/ExpiryNotification/TruncatedAddress.js (hunk 15 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/TruncatedAddress.js]: Replaced w. styled-component

File: src/components/ExpiryNotification/ExpiryNotificationModal.js (hunk 14 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js]: Replaced w. styled component

File: src/components/ExpiryNotification/ExpiryNotificationModal.js (hunk 13 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js]: Removed inline css & hoisted `font-weight` to parent styled component

File: src/components/Address/Address.js (hunk 12 above)
Comment (mjpowersjr) [reply src/components/Address/Address.js]: Removed inline css + replaced w. new 'actions' grid item approach to resolve wrapping issue that @makoto discovered. See `Address.js` & `RenewAll.js` for details in:

* https://github.com/ensdomains/ens-app/pull/776/commits/f88810d398bfe4ad49d2a55897ba20fdab07d9c4
* https://github.com/ensdomains/ens-app/pull/776/commits/fec5670394badd68c8389674a4fa568873868cca

File: src/components/Calendar/Calendar.js (hunk 11 above)
Comment (mjpowersjr) [reply src/components/Calendar/Calendar.js]: Updated

File: src/components/ExpiryNotification/ExpiryNotificationModal.js (hunk 10 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js]: Removed

File: src/components/ExpiryNotification/ExpiryNotificationModal.js (hunk 9 above)
Comment (mjpowersjr) [reply src/components/ExpiryNotification/ExpiryNotificationModal.js]: * close - Removed this completely after prior discussion w. @makoto 
* placeholder - moved  to `expiryNotification.placeholder`- https://github.com/ensdomains/ens-app/pull/776/commits/481c5e84c885cccccad7408b4893a25dc48444c8

**Comment by mjpowersjr** [comment]
> One more thing.
> 
> In mobile view, I cannot see input field border.
> 
> Screenshot 2020-05-26 at 21 34 23

I believe this is fixed, but if you could check on your browser it would be appreciated. :)

**Comment by mjpowersjr** [comment]
Ok, I think that covers all the changes from round 1 of the review. Thanks for all the detailed feedback! :-)

My apologies for missing the renewal box wrapping issue before initial PR. I'm embarrassed that I let that slip through.

**Comment by makoto** [comment]
> I believe this is fixed, but if you could check on your browser it would be appreciated. :)

I checked on chrome but still seeing the same issue (looks fine on Opera).

**Comment by makoto** [comment]
Looks like alignment seems broken

Screenshot 2020-05-28 at 13 28 40

**Comment by mjpowersjr** [comment]
> Looks like alignment seems broken
> 
Ok, I was able to tweak the breakpoint for the select all padding. This seem to help with screen sizes down to 768px. Smaller than 768 we run into alignment issues, but the behavior matches what's in production today.

**Comment by mjpowersjr** [comment]
> > I believe this is fixed, but if you could check on your browser it would be appreciated. :)
> 
> I checked on chrome but still seeing the same issue (looks fine on Opera).

Ok, I adjusted the input style to match that of the record creation input form. If that does not work, I'm not sure where to go next - I have a consistent border across firefox, chrome, opera, and brave on my linux box. If this change doesn't work, do you mind taking a quick look?

Thanks!

**Comment by makoto** [comment]
border of email input on mobile chrome seems fixed, but post submit popup does not seem to have email icon

Screenshot 2020-05-28 at 21 14 48


Screenshot 2020-05-28 at 21 17 01

**Comment by makoto** [comment]
Possibly the icon is disappearing on narrower window or mobile window

**Comment by rliebert90** [comment]
I think it's fine to hide the icon on mobile. But if anyone feels strongly about including it, we could put it above the text and increase the height of the popup.

**Pull request merged**